docker-compose-rule-core/src/main/java/com/palantir/docker/compose/connection/waiting/ClusterWaitInterface.java

```diff
@@ -0,0 +1,23 @@
+/*
+ * (c) Copyright 2019 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.docker.compose.connection.waiting;
+
+import com.palantir.docker.compose.connection.Cluster;
+
+public interface ClusterWaitInterface {
```

Comment:
I had to extract out this interface, however `ClusterWait` needs to remain the same. Open to a better name.

Reply:
why did we have to extract this? it doesn't look like there are other implementors except `ClusterWait`

Reply:
There's a sneaky one here: https://github.com/palantir/docker-compose-rule/pull/292/files#diff-fd69676598fa179a6d84c0bd468b2eeeR85

Reply:
Actually, you're right, that doesn't even need to be an interface any more. At one put I was adding these things in the builder, now that's no longer happening it should be fine.